File: 002_Add_Two_Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* res = new ListNode(0);
        ListNode* head = res;
        int jinwei = 0;
        while (l1 != NULL && l2 != NULL) {
            int tmp = (l1->val + l2->val + jinwei) % 10;
            head->next = new ListNode(tmp);
            jinwei = (l1->val + l2->val + jinwei) / 10;
            head = head->next;
            l1 = l1->next;
            l2 = l2->next;
        }
        while (l1 != NULL) {
            head->next = new ListNode((l1->val + jinwei) % 10);
            jinwei = (l1->val + jinwei) / 10;
            head = head->next;
            l1 = l1->next;
        }
        while (l2 != NULL) {
            head->next = new ListNode((l2->val + jinwei) % 10);
            jinwei = (l2->val + jinwei) / 10;
            head = head->next;
            l2 = l2->next;
        }
        if (jinwei != 0) {
            head->next = new ListNode(jinwei);
        }
        return res->next;
    }
};

File: 002_Add_Two_Numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        res = ListNode(0)
        head = res
        jinwei = 0
        while l1 and l2:
            head.next = ListNode((l1.val + l2.val + jinwei) % 10)
            jinwei = (l1.val + l2.val + jinwei) / 10
            l1 = l1.next
            l2 = l2.next
            head = head.next
        while l1:
            head.next = ListNode((l1.val + jinwei) % 10)
            jinwei = (l1.val + jinwei) / 10
            l1 = l1.next
            head = head.next
        while l2:
            head.next = ListNode((l2.val + jinwei) % 10)
            jinwei = (l2.val + jinwei) / 10
            l2 = l2.next
            head = head.next
        if jinwei != 0:
            head.next = ListNode(jinwei)
        return res.next
         
        